app/services/sign_in/refresh_token_decryptor.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module SignIn
+  class RefreshTokenDecryptor
+    attr_reader :encrypted_refresh_token
+
+    def initialize(encrypted_refresh_token:)
+      @encrypted_refresh_token = encrypted_refresh_token
+    end
+
+    def perform
+      split_refresh_token = split_encrypted_refresh_token(encrypted_refresh_token)
+
+      decrypted_part = get_decrypted_part(split_refresh_token)
+      nonce = get_nonce_part(split_refresh_token)
+      version = get_version_part(split_refresh_token)
```

Comment:
Thoughts on making `split_refresh_token` a private method that `get_decrypted_part` calls directly (instead of being passed)?

`get_` could be removed from the methods and these locals would go away and `validate_token!` would call the private methods instead.

Reply:
I reorganized this a bit to hopefully make this appear simpler